providers/task.rb

```diff
@@ -181,6 +185,43 @@ def task_need_update?
     @current_resource.user != @new_resource.user
 end
 
+def set_cwd(folder)
+  Chef::Log.debug 'looking for existing tasks'
+
+  # we use shell_out here instead of shell_out! because a failure implies that the task does not exist
+  task_xml = shell_out("schtasks /Query /TN \"#{@new_resource.task_name}\" /XML").stdout
+  doc = REXML::Document.new(task_xml)
+
+  Chef::Log.debug 'Removing former CWD if any'
+  doc.root.elements.delete('Actions/Exec/WorkingDirectory')
+
+  unless folder.nil?
+    Chef::Log.debug 'Setting CWD as #folder'
+    cwd_element = REXML::Element.new('WorkingDirectory')
+    cwd_element.add_text(folder)
+    exec_element = doc.root.elements['Actions/Exec']
+    exec_element.add_element(cwd_element)
+  end
+
+  temp_task_file = ::File.join(ENV['TEMP'], 'windows_task.xml')
```

Comment:
`Tempfile` does not work out here because we need to give `schtasks` a path to read from and cant give it the actual Tempfile. So then `schedtysks` complains about the file being open.

I feel like this is safe enough though since it uses the temp directory and deletes it in an `ensure` block.